content/intro-to-storybook/ember/en/simple-component.md

```diff
@@ -0,0 +1,223 @@
+
+`action()` allows us to create a callback that appears in the **actions** panel of the Storybook UI when clicked. So when we build a pin button, we’ll be able to determine in the test UI if a button click is successful.
+
+As we need to pass the same set of actions to all permutations of our component, it is convenient to bundle them into a single `actionData` variable and pass them into our story definition each time (where they accessed when triggered!!!).
+
+Another nice thing about bundling the actions into `actionsData` is that you can `export` that variable and use the actions in stories for components that reuse this component, as we'll see later.
+
+When creating a story we use a base task (`taskData`) to build out the shape of the task the component expects. This is typically modelled from what the true data looks like. Again, `export`-ing this shape will enable us to reuse it in later stories, as we'll see.
+
+<div class="aside">
+<a href="https://storybook.js.org/addons/introduction/#2-native-addons"><b>Actions</b></a> help you verify interactions when building UI components in isolation. Oftentimes you won't have access to the functions and state you have in context of the app. Use <code>action()</code> to stub them in.
+</div>
+
+## Config
+
+We'll also need to make one small change to the Storybook configuration so it notices our recently created stories. Change your configuration file (`.storybook/main.js`) to the following:
+
+```javascript
+// .storybook/main.js
+
+module.exports = {
+  stories: ['../app/components/**/*.stories.js'],
+  addons: ['@storybook/addon-actions', '@storybook/addon-links'],
+};
+```
+
+Once we’ve done this, restarting the Storybook server should yield test cases for the three Task states:
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/intro-to-storybook//inprogress-task-states.mp4"
+    type="video/mp4"
+  />
+</video>
+
+## Build out the states
+
+Now we have Storybook setup, styles imported, and test cases built out, we can quickly start the work of implementing the HTML of the component to match the design.
+
+The component is still basic at the moment. First write the code that achieves the design without going into too much detail:
+
+```handlebars
+
+{{!--  app/components/Task.hbs --}}
+<div class="list-item {{@task.state}}">
+  <label class="checkbox">
+    <input type="checkbox" disabled name="checked" checked={{this.isTaskArchived}}/>
+    <span class="checkbox-custom" onclick={{this.handleArchiveTask}}/>
+  </label>
+  <div class="title">
+    <input type="text" readonly value={{@task.title}} placeholder="Input title" />
+  </div>
+  <div class="actions">
+    {{#if (not-eq @task.state "TASK_ARCHIVED")}}
+      <a href="/" onclick={{this.handlePinTask}}><span class="icon-star"  /></a>
```

Comment:
```suggestion
      <a href="/" {{on "click" (stop-propagation (prevent-default (fn @pinTask @task.id)))}}><span class="icon-star"  /></a>
```

I used `(stop-propagation)` and `(prevent-default)` from [`ember-event-helpers`](https://github.com/buschtoens/ember-event-helpers) package. Or just use a `<span>` here as everywhere else and ignore them.

Reply:
@gossi i was checking this and why add another addon when we have a means to address this natively? Also i'm aiming for consistency throughout all the versions of the documentation.